src/components/PopupWithForm.js

```diff
@@ -17,6 +18,14 @@ export default class PopupWithForm extends Popup {
     return formInputValue;
   }
 
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = "Save";
```

Comment:
Needs correction: You should return the initial button text here

One of the options to fix that would be adding parameters `buttonText` and `loadingButtonText` to the constructor of `PopupWithForm` and using them to change button text. 

```js
constructor(popupSelector, { handleSubmit, buttonText, loadingButtonText }) { 
  …
  this._buttonText = buttonText;
  this._loadingButtonText = loadingButtonText;
}

...
this._submitButton.textContent = this._buttonText
```

Reply:
Would I pass what I want the submit button text content to be as a parameter when I instantiate the PopupWithForm class in index.js?

Reply:
Disregard this question it took me a bit but I figured it out.